webpack/assets/javascripts/react_app/components/ModelsTable/ModelsTable.js

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Table as PfTable } from 'patternfly-react';
+
+class ModelsTable extends React.Component {
+  componentWillMount() {
+    this.props.getModels(this.props.url);
+  }
+  render() {
+    const {
+      columns, filterFnc, results, store,
+    } = this.props;
+    return (
+      <PfTable.PfProvider
```

Comment:
There is also a foreman wrapper of the patternfly table that provides some additional functionality (such as integrating with the `EmptyState` component) in @amirfefer's about page PR (https://github.com/theforeman/foreman/pull/5024).

Reply:
Yes, I saw that PR. Definitely, will be useful when I turn `Hardware Models` into a `React` page. Although, we need to figure out best practices with `reactabular` since there are many ways to setup a table and each one of us did it differently due to different requirements/needs.

Reply:
I would hope we have a common set of defaults that is mostly plug&pray (similar to how table css classes are defined today)